internal/pkg/application/transform/transformers.go

```diff
@@ -241,71 +353,117 @@ func WaterConsumptionObserved(ctx context.Context, msg iotcore.MessageAccepted,
+	toLtr := func(m3 float64) float64 {
+		return math.Floor((m3 + 0.0005) * 1000)
+	}
 
-	// first create WaterConsumptionObserved for first logValue
-	if logVolume, ok := msg.GetFloat64("LogVolume"); ok {
-		if logDateTime, ok := msg.GetTime("LogDateTime"); ok {
-			vol := math.Floor((logVolume + 0.0005) * 1000)
-			dt := time.Unix(int64(logDateTime), 0).UTC().Format(time.RFC3339Nano)
-			props := waterConsumptionProps(vol, dt, observedBy, msg)
+	toDateStr := func(t float64) string {
+		return time.Unix(int64(t), 0).UTC().Format(time.RFC3339Nano)
+	}
 
-			err := storeWaterConsumption(ctx, log, cbClient, entityID, props...)
+	for _, rec := range msg.Pack {
+		if rec.Name == "1" {
+			w := Number("waterConsumption", toLtr(*rec.Sum), p.UnitCode("LTR"), p.ObservedAt(toDateStr(rec.Time)), p.ObservedBy(observedBy))
+			p := append(props, w)
+			err := storeWaterConsumption(ctx, log, cbClient, entityID, p...)
 			if err != nil {
 				return err
 			}
 		}
 	}
 
-	// then continue with delta volumes
-	for _, record := range msg.Pack {
-		if strings.EqualFold("DeltaVolume", record.Name) {
-			vol := math.Floor((*record.Sum + 0.0005) * 1000)
-			dt := time.Unix(int64(record.Time), 0).UTC().Format(time.RFC3339Nano)
-			props := waterConsumptionProps(vol, dt, observedBy, msg)
+	return nil
+}
 
-			err := storeWaterConsumption(ctx, log, cbClient, entityID, props...)
-			if err != nil {
-				return err
-			}
+func GreenspaceRecord(ctx context.Context, msg core.MessageAccepted, cbClient client.ContextBrokerClient) error {
+	curDateTime := msg.Timestamp
+	if cdt, ok := msg.GetString("CurrentDateTime"); ok {
+		if idx := strings.Index(cdt, "."); idx > 0 {
+			curDateTime = cdt[0:idx] + "Z"
 		}
 	}
 
-	// last add current volume if exists
-	if currentVolume, ok := msg.GetFloat64("CurrentVolume"); ok {
-		if currentDateTime, ok := msg.GetTime("CurrentDateTime"); ok {
-			vol := math.Floor((currentVolume + 0.0005) * 1000)
-			dt := time.Unix(int64(currentDateTime), 0).UTC().Format(time.RFC3339Nano)
-			props := waterConsumptionProps(vol, dt, observedBy, msg)
+	entityID := fmt.Sprintf("%s%s", "urn:ngsi-ld:GreenspaceRecord:", msg.Sensor)
+	observedBy := fmt.Sprintf("%s%s", fiware.DeviceIDPrefix, msg.Sensor)
+
+	headers := map[string][]string{"Content-Type": {"application/ld+json"}}
 
-			err := storeWaterConsumption(ctx, log, cbClient, entityID, props...)
+	buildfragment := func(patchProperties ...entities.EntityDecoratorFunc) error {
+		fragment, err := entities.NewFragment(patchProperties...)
+		if err != nil {
+			return fmt.Errorf("entities.NewFragment failed: %w", err)
+		}
+
+		properties := append(patchProperties, DateObserved(curDateTime))
+
+		//_, err = cbClient.UpdateEntityAttributes(ctx, entityID, fragment, headers)
+		_, err = cbClient.MergeEntity(ctx, entityID, fragment, headers)
+
+		if err != nil {
+			// If we failed to update the entity's attributes, we need to create it
```

Comment:
check here if the error is that we can't find the entity, as we do it in IndoorEnvironmetObserved above. we've used that same pattern in another service as well. Line 202 onwards.